[inductor] Disable two tests on H100 due to a ptxas bug exposed by recent Triton pin update.

htyu opened the pull request:
Summary: Recent Triton pin upgrade includes a LLVM pin update that exposes a PTXAS optimizer issue causing two tests failed. The tests pass with `DISABLE_PTXAS_OPT=1`.

Test Plan:
buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/inductor:test_inductor_cuda -- --exact 'caffe2/test/inductor:test_inductor_cuda - test_upsample_nearest2d_backward_cuda (caffe2.test.inductor.test_torchinductor.GPUTests)'

   buck2 test 'fbcode//mode/opt' fbcode//hammer/modules/sequential/encoders/tests:hstu_transducer_cint_tests -- --exact 'hammer/modules/sequential/encoders/tests:hstu_transducer_cint_tests - test_inference_fx_ts_compatibility (hammer.modules.sequential.encoders.tests.hstu_transducer_cint_tests.HSTUTransducerCIntFXTStest_0)'

Differential Revision: D58532831


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

amjames commented on test/inductor/test_torchinductor.py:
```diff
@@ -5756,6 +5756,10 @@ def fn(a):
 
         self.common(fn, (torch.randn([2, 4, 37, 38, 39]),))
 
+    @unittest.skipIf(
```
IMO xfail might be more appropriate here. On a future pin update when the issue is resolved we will get signal that the test is passing.

htyu replied:
Good point. I could use `@unittest.expectedFailure` but didn't figure out how to set it conditionally. Do you have any pointers? Thanks.

amjames replied:
you have to write your own decorator, we hav lots of them but I don't think there is one for this yet. 


```python
def xfailIfSM90(fn):
    if torch.cuda.is_available() and torch.cuda.get_device_capability() == (9, 0):
        return unittest.expectedFailure(fn)
	return fn
```

bertmaher reviewed (approved):
This is kind of unfortunate but I guess it's okay for now.  Do you have any insight into what changed in Triton to trigger the ptxas bug?

htyu commented:
> This is kind of unfortunate but I guess it's okay for now. Do you have any insight into what changed in Triton to trigger the ptxas bug?

It's due to llvm pin update.

htyu committed:
[inductor] Disable two tests on H100 due to a ptxas bug exposed by recent Triton pin update. (#128625)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/128625

Recent Triton pin upgrade includes a LLVM pin update that exposes a PTXAS optimizer issue causing two tests failed. The tests pass with `DISABLE_PTXAS_OPT=1`.

Test Plan:
buck2 test 'fbcode//mode/opt' fbcode//caffe2/test/inductor:test_inductor_cuda -- --exact 'caffe2/test/inductor:test_inductor_cuda - test_upsample_nearest2d_backward_cuda (caffe2.test.inductor.test_torchinductor.GPUTests)'

   buck2 test 'fbcode//mode/opt' fbcode//hammer/modules/sequential/encoders/tests:hstu_transducer_cint_tests -- --exact 'hammer/modules/sequential/encoders/tests:hstu_transducer_cint_tests - test_inference_fx_ts_compatibility (hammer.modules.sequential.encoders.tests.hstu_transducer_cint_tests.HSTUTransducerCIntFXTStest_0)'

Reviewed By: bertmaher, minjang, chenyang78

Differential Revision: D58532831
